File: SEP490_Ezstay/AccountAPI/Service/Interfaces/IUserService.cs
using AccountAPI.Data;
using AccountAPI.DTO.Reponse;
using AccountAPI.DTO.Request;
using AccountAPI.DTO.Resquest;
using System.Security.Claims;

namespace AccountAPI.Service.Interfaces
{
    public interface IUserService
    {
        Task<bool> CreateProfileAsync(Guid userId, UserDTO userDto);
        Task<UserResponseDTO?> GetProfileAsync(Guid userId);
        Task<bool> UpdateProfile(Guid userId, UpdateUserDTO userDto);
        Task<bool> UpdatePhoneAsync(Guid userId, string newPhone);
        Task<bool> VerifyPhoneOtpAsync(string phone, string otp);
        Task<bool> SendOtpToPhoneAsync(string phone);
        Task<bool> UpdateEmailAsync(string currentEmail, string newEmail, string otp);
        Task<UserResponseDTO> GetPhone(string phone);
        Task<UserResponseDTO> GetCitizenIdNumber(string citizenIdNumber);
        Task<bool> CheckProfileAsync(Guid id);



    }

}

File: SEP490_Ezstay/AccountAPI/AccountAPI/Service/Interfaces/IUserService.java
package AccountAPI.Service.Interfaces;

import AccountAPI.DTO.Request.UserDTO;
import AccountAPI.DTO.Reponse.UserResponseDTO;
import AccountAPI.DTO.Request.UpdateUserDTO;

public interface IUserService {

	public Task<boolean> CreateProfileAsync(Guid aUserId, UserDTO aUserDto);

	public Task<?UserResponseDTO> GetProfileAsync(Guid aUserId);

	public Task<boolean> UpdateProfileAsync(Guid aUserId, UpdateUserDTO aDto);

	public Task<boolean> UpdatePhoneAsync(Guid aUserId, String aNewPhone);

	public Task<boolean> VerifyPhoneOtpAsync(String aPhone, String aOtp);

	public Task<boolean> SendOtpToPhoneAsync(String aPhone);

	public Task<boolean> UpdateEmailAsync(String aCurrentEmail, String aNewEmail, String aOtp);
}